src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/MenuItemAutomationPeer.cs

```diff
@@ -316,6 +316,27 @@ internal void RaiseExpandCollapseAutomationEvent(bool oldValue, bool newValue)
                 newValue ? ExpandCollapseState.Expanded : ExpandCollapseState.Collapsed);
         }
 
+        internal void RaiseToggleStatePropertyChangedEvent(bool oldValue, bool newValue)
+        {
+            if (oldValue != newValue)
+            {
+                RaisePropertyChangedEvent(TogglePatternIdentifiers.ToggleStateProperty, ConvertToToggleState(oldValue), ConvertToToggleState(newValue));
+            }
+        }
+
+        private static ToggleState ConvertToToggleState(bool? value)
+        {
+            switch (value)
+            {
+                case (true):
+                    return ToggleState.On;
+                case (false):
+                    return ToggleState.Off;
+                default:
+                    return ToggleState.Indeterminate;
+            }
+        }
```

Comment:
Small suggestion:
ConvertToToggleState is private and is always used with a non-nullable bool for the parameter `value`. Converting a non-nullable bool -> nullable bool -> non-nullable bool is slower than just passing a non-nullable for the parameter `value` since no conversion occurs.
```suggestion
     private static ToggleState ConvertToToggleState(bool value)
         => value ? ToggleState.On : ToggleState.Off;
```

Reply:
Thanks for the suggestion, I have incorportated the changes.